### ISSUE TITLE
Bug - [Select/FormSelect] - [Expanded menu is in white theme not dark (regression)]

### ISSUE THREAD

**Issue by jelly** [patternfly/patternfly#5695]
**Describe the problem**

We noticed this when upgrading from 53 to 98 alpha release and can reproduce it on the PF5 [demo website](https://pf5.patternfly.org/components/forms/form-control)

![image](https://github.com/patternfly/patternfly/assets/67428/3eb24240-b0eb-47a2-a9a8-8df2029b5373)

**How do you reproduce the problem?**

1. Go to https://pf5.patternfly.org/components/forms/form-control
2. Enable dark theme
3. Open any of the dropdown menu's.

**Expected behavior**

Dropdown menu should be dark

**Is this issue blocking you?**
List the workaround if there is one.

**Screenshots**
If applicable, add screenshots to help explain the issue.

**What is your product and what release date are you targeting?**

Cockpit

**Comment by jelly** [comment]
This actually seems to be mostly a FormSelect issue or well, where `select` is used, previously the select had a `class="pf-v5-c-form-control"` and with the new version it does not as it's now a child of a div with that class.

**Comment by jelly** [comment]
This was previously overridden by patternfly, but as the select now lacks the `pf-v5-c-form-control` class the css rules no longer apply.

**Comment by jelly** [comment]
Re-testing with https://patternfly-react-v5.surge.sh/components/forms/form-select makes it seem that this is Chrome(ium) only.

![image](https://github.com/patternfly/patternfly/assets/67428/db5ec41f-2eba-49e9-b908-9b2720ade8a8)

![image](https://github.com/patternfly/patternfly/assets/67428/8d92c582-916b-4860-8e7c-f08cdd8bafe6)

**Comment by mcoker** [comment]
@jelly I don't have chromium installed, but checking in chrome and chrome canary, I'm not able to reproduce what you have in the screenshot. The native `<select>` menu is taking on whatever the OS preference is for dark/light theme, regardless the setting for dark/light mode in patternfly. I confirmed this is the case in our latest pre-release version of PF as well as the version prior to the update that wrapped the `<select>` in a div. Is this only happening on Chromium?

**Comment by jelly** [comment]
I just spun up a Windows VM with Edge and it shows as this. So I didn't set the OS preference to dark theme but only toggled the radio button in the PatternFly website.


![image](https://github.com/patternfly/patternfly/assets/67428/2a283494-61cf-49ec-8e03-e97adac4696d)

**Comment by garrett** [comment]
Chrome doesn't have proper support for system themes. This has been an issue for several years, especially in (but not limited to) Linux.

Therefore, it _still_ needs to really be fully overridden with both foregrand and background colors (not just one) — which is a good thing to do anyway (that is: if you set a color, you need to set a background, and vice versa).

Chrome _finally_ mostly released support in version 114 (around a month and a half ago), but it's relatively recent and not all Chrome forks support it yet. And depending on the desktop and distribution method, the setting might not be properly exposed to Chrome, Chromium, one of the browser forks, and even Electron apps, and it could even do something different in X11 and Wayland (as they're different renderers).

Here's Chrome from Flatpak on Fedora 38 on GNOME on Wayland:

![image](https://github.com/patternfly/patternfly/assets/10246/c144805b-ae17-44e4-8b5d-6b0f349e9452)

Same for Chromium:

![image](https://github.com/patternfly/patternfly/assets/10246/34e3a541-03e0-4d64-a1bb-b82b583f8596)

You can see how it's incomplete as the browser "chrome" (UI) on Chromium is still stuck in light mode whereas everything else is in dark (including most of Cockpit):

![image](https://github.com/patternfly/patternfly/assets/10246/022af15e-d074-4dfa-99bd-8f852a67314c)

(For reference, Firefox and GNOME Web (Safari on Linux, basically) have both done the right thing for years. This is one of the many side effects of Google controlling the engine and not doing the right thing on both Linux and Windows... possibly even macOS, but I cannot confirm that.)

While this is a problem when using Chrome's buggy implementation of dark mode support (which is better than nothing), it's _also_ a problem with PatternFly. It really needs to set both the background and foreground color. Depending on OS settings is not fully reliable, and relying on OS settings is incorrect in this instance anyway, as PatternFly allows for dark and light mode at the app level (which Cockpit also supports) — and widgets should all be consistently dark, light, or auto, based on the setting. The default color shouldn't "leak" through when there's a mismatch (either due to a manual in-app setting or a semi-broken implementation of dark mode).

---

TL;DR: Set foreground (`color`) and background (`background`) for widgets together, not one separate from the other. And both should be set (_somewhere_, not necessarily on every widget) to make sure dark and light modes are consistent.

**Comment by garrett** [comment]
This is closed, but it wasn't fixed.

Here's a screenshot of Chrome on PatternFly.org, in dark mode using the dark toggle, as of today (2023-12-11):

![dropdown showing the bug](https://github.com/patternfly/patternfly/assets/10246/85b15fc6-d1e7-4799-86f1-932f933e23a1)
